src/store/reducers/article.js

```diff
@@ -5,81 +5,87 @@ const initiaState = {
-            return {
-                ...state,
-                loading: true,
-                error:null,
-                article:null,
-                message: ""
-            };
-        case actionTypes.GET_ARTICLE_FAIL:
-            return {
-                ...state,
-                loading: false,
-                article: null,
-                error: action.error
-            };
-        case actionTypes.GET_ARTICLE_SUCCESS:
-            return {
-                ...state,
-                loading: false,
-                article: action.data.articles,
-                error: null,
-                averageRating: action.data.articles.average_ratings
-            };
-        case actionTypes.DELETE_ARTICLE:
-            return {
-                ...state,
-                loading: false,
-                article: action.data.articles,
-                error: null
-            };
-        case actionTypes.CREATE_ARTICLE_SUCCESS:
-            return {
-                ...state,
-                loading: false,
-                article: action.data.articles,
-                error: null,
-                message: "Article created successfully"
-            };
-        case actionTypes.CREATE_ARTICLE_FAIL:
-            return {
-                ...state,
-                loading: false,
-                article: null,
-                error: action.error
-            };
-        case actionTypes.UPDATE_ARTICLE_SUCCESS:
-            return {
-                ...state,
-                loading: false,
-                article: action.data.articles,
-                error: null,
-                message: "Changes saved succefully"
-            };
-        case actionTypes.UPDATE_ARTICLE_FAIL:
-            return {
-                ...state,
-                loading: false,
-                article: null,
-                error: action.error
-            };
-        case actionTypes.ACTION_ENDED:
-            return {
-                ...state,
-                article: null,
-                error: null,
-                loading: false,
-                message: ""
-            };
-        default: return state;
-    };
+    case actionTypes.GET_ARTICLE_START:
+        return {
+            ...state,
+            loading: true,
+            error:null,
+            article:null,
+            message: ""
+        };
+    case actionTypes.GET_ARTICLE_FAIL:
+        return {
+            ...state,
+            loading: false,
+            article: null,
+            error: action.error
+        };
+    case actionTypes.GET_ARTICLE_SUCCESS:
+        return {
+            ...state,
+            loading: false,
+            article: action.data.articles,
+            error: null,
+            averageRating: action.data.articles.average_ratings,
+            facebook:action.data.articles.share_links.facebook,
+            twitter:action.data.articles.share_links.twitter,
+            email:action.data.articles.share_links.email
```

Comment:
@missvicki I feel like this is a little bit not lean. I mean could I get a clear reason to why you decided to add twitter, email and Facebook states in the reducer. I mean the article state that was initially set was containing all this and all you had to do is just desctructre it and get all the information you needed.

Reply:
Thank you for paying attention to this. I am working on this.